src/Autofac/Features/ResolveAnything/AnyConcreteTypeNotAlreadyRegisteredSource.cs

```diff
@@ -91,16 +91,39 @@ public AnyConcreteTypeNotAlreadyRegisteredSource(Func<Type, bool> predicate)
                 return Enumerable.Empty<IComponentRegistration>();
             }
 
+            // Check to resolve issue #925
```

Comment:
Check what to resolve issue #925? Historically, we've actually moved from Google Code to GitHub and we've transferred issues around so just an issue number isn't enough context. Ideally comments here will explain not only the issue but _why_ we're doing what we're doing.

For example, the comment here probably doesn't make sense. This `if` check didn't change for #925.

However, just below where we exclude `Lazy<>` - why? A comment there explaining `Lazy<T>` has a default no-parameter constructor and potentially interferes with the `Lazy<T>` handling provided elsewhere... that's important. It also sort of explains why `Func<>` is _not_ excluded... but maybe it should be?

Reply:
I broke out the main body of the code I added into its own private method so that I could document it more clearly. I added some single line comments that I hope are helpful also.

As for Func<>, I don't think I need to exclude it explicitly since ACTNARS already excludes all sub-types of Delegate.